sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py

```diff
@@ -311,6 +307,11 @@ def send(self, message, timeout=None):
                 :caption: Send message.
 
         """
+        if isinstance(message, BatchMessage):
+            if len(message) == 0:
+                raise MessageError("A BatchMessage must have at least one Message")
```

Comment:
Is this the only place we're throwing MessageError?  I don't love the naming regardless, it seems like ValueError would be more appropriate.

Reply:
MessageError is also thrown when you try to send a message twice, the second send will get uamqp.errors.MessageAlreadySettled (I know it's being discussed that we should allow a message be sent twice, but now it still raises a uamqp exception so I wrapped it to MessageError). When you send a too large message, the service also returns an error. In summary, a MessageError means the message itself is not in a good state to be successfully sent out.

For this case, I was also hesitating between ValueError and MessageError.

Reply:
Changed to raise MessageContentTooLarge as per our discussion